runtime/onert/backend/trix/DevContext.cc

```diff
@@ -0,0 +1,265 @@
+  //      be terminated
+  _batch_thread_pool.reset(nullptr);
+
+  for (const auto &dev_handle : _dev_handles)
+  {
+    unregisterNPUmodel_all(dev_handle);
+    putNPUdevice(dev_handle);
+  }
+}
+
+ModelID DevContext::registerModel(const std::string &model_file_path)
+{
+  auto meta = getNPUmodel_metadata(model_file_path.c_str(), false);
+
+  if (meta == nullptr)
+  {
+    throw std::runtime_error("Unable to extract the model metadata");
+  }
+
+  generic_buffer file_info;
+  file_info.type = BUFFER_FILE;
+  file_info.filepath = model_file_path.c_str();
+  file_info.size = meta->size;
+
+  ModelID model_id;
+
+  for (uint32_t dev_num = 0; dev_num < _dev_handles.size(); ++dev_num)
+  {
+    // Register model for each device
+    uint32_t model_id_at_device;
+    if (registerNPUmodel(_dev_handles.at(dev_num), &file_info, &model_id_at_device) < 0)
+    {
+      throw std::runtime_error("Failed to register npu model");
+    }
+
+    if (dev_num == 0)
+    {
+      model_id = model_id_at_device;
+      _meta_map[model_id_at_device] = std::shared_ptr<npubin_meta>(meta);
+    }
+    else
+    {
+      _meta_map[model_id_at_device] = _meta_map[model_id];
+    }
+
+    _model_ids[model_id].resize(dev_num + 1);
+    _model_ids[model_id].at(dev_num) = model_id_at_device;
+  }
+
+  // Return the model id for device 0 only
+  return model_id;
+}
+
+void DevContext::unRegisterModel(ModelID model_id)
+{
+  for (uint32_t dev_num = 0; dev_num < _dev_handles.size(); ++dev_num)
+  {
+    const auto model_id_at_device = _model_ids.at(model_id).at(dev_num);
+    const auto &dev_handle = _dev_handles.at(dev_num);
+
+    // Remove meta data
+    _meta_map.erase(model_id_at_device);
+
+    // Unregister Model for each device
+    unregisterNPUmodel(dev_handle, model_id_at_device);
+  }
+  // Remove model IDs
+  _model_ids.erase(model_id);
+}
+
+void DevContext::requestRun(ModelID model_id, input_buffers *input_bufs, tensors_data_info *in_info,
+                            output_buffers *output_bufs, tensors_data_info *out_info,
+                            size_t batch_size)
+{
+  if (batch_size > 1)
+  {
+    if (in_info->num_info != 1)
+    {
+      throw std::runtime_error("Supported only an input that has batch now");
+    }
+    if (out_info->num_info != 1)
+    {
+      throw std::runtime_error("Supported only one output now");
+    }
+
+    // inputs/outputs for each batch
+    std::vector<input_buffers> in_buffers_vec(batch_size);
+    std::vector<output_buffers> out_buffers_vec(batch_size);
+
+    std::vector<tensors_data_info> in_infos(batch_size);
+    std::vector<tensors_data_info> out_infos(batch_size);
+
+    // Run on thread pool
+    std::vector<std::future<int32_t>> batch_futures;
+    for (uint32_t batch_num = 0; batch_num < batch_size; ++batch_num)
+    {
+      // Set buffers of inputs/outputs for each batch
+      // TODO Support multiple inputs/outputs
```

Comment:
Why don't we support multiple intpus/ouptuts by updating setBufferByBatch ?

Reply:
We're not sure which of the multiple inputs/outputs are affected by batch. So for now it would be better to set each buffer.

Reply:
Then, let's revise this after inputs/outputs spec is defined precisely.